### ISSUE TITLE
Move documentation out of wiki

### ISSUE THREAD

**Issue by kal** [dotnetrdf/dotnetrdf#342]
One big issue with using the wiki for documentation is that it is not possible to keep multiple versions easily accessible to users. So the wiki always reflects the "latest docs" (which may or may not reflect the actual latest release). Another issue is that links to the API docs then go out of the wiki to the dotnetrdf.org site (again this site only hosts the latest API docs).

The ideal solution to this might be using readthedocs.org to build and host the documentation but it doesn't look like their default build environment supports the necessary tooling for integrating docfx API documentation into the build. Another possibility might be to use the same toolchain as readthedocs.org (Sphinx + rtd templates) but run the build as a GitHub Action / AppVeyor build and host the docs on the dotnetrdf.org site.

Probably needs a bit of investigation of whats doable with rtd.org first.

**Comment by kal** [comment]
Also might e worth looking at what docfx could do (https://dotnet.github.io/docfx/)

**Comment by kal** [comment]
The solution I've ended up with is a separate repo (https://github.com/dotnetrdf/docs) which contains DocFX projects for building the user docs and API docs as an integrated set. The current build process is set up so that we can have one copy of the docs per minor release and the built docs can be simply copied over to our GitHub Pages site.

The wiki pages on in this repo could maybe get removed, but for now I've settled for just putting a note on the home page of the wiki.
